SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/SpatialMultiWorkerSettings.cpp

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "LoadBalancing/SpatialMultiWorkerSettings.h"
+
+#include "LoadBalancing/AbstractLBStrategy.h"
+#include "LoadBalancing/GridBasedLBStrategy.h"
+#include "LoadBalancing/LayeredLBStrategy.h"
+#include "LoadBalancing/OwnershipLockingPolicy.h"
+#include "Utils/LayerInfo.h"
+
+#include "Misc/MessageDialog.h"
+
+#define LOCTEXT_NAMESPACE "SpatialMultiWorkerSettings"
+
+#if WITH_EDITOR
+void UAbstractSpatialMultiWorkerSettings::PostEditChangeProperty(struct FPropertyChangedEvent& PropertyChangedEvent)
+{
+	Super::PostEditChangeProperty(PropertyChangedEvent);
+
+	const FName Name = (PropertyChangedEvent.MemberProperty != nullptr) ? PropertyChangedEvent.MemberProperty->GetFName() : NAME_None;
+
+	if (Name == GET_MEMBER_NAME_CHECKED(UAbstractSpatialMultiWorkerSettings, WorkerLayers))
+	{
+		ValidateFirstLayerIsDefaultLayer();
+		ValidateNonEmptyWorkerLayers();
+		ValidateSomeLayerHasActorClass();
+		ValidateNoActorClassesDuplicatedAmongLayers();
+		ValidateAllLayersHaveUniqueNonemptyNames();
+		ValidateAllLayersHaveLoadBalancingStrategy();
+	}
+	else if (Name == GET_MEMBER_NAME_CHECKED(UAbstractSpatialMultiWorkerSettings, LockingPolicy))
+	{
+		ValidateLockingPolicyIsSet();
+	}
+};
+#endif
+
+uint32 UAbstractSpatialMultiWorkerSettings::GetMinimumRequiredWorkerCount() const
+{
+	uint32 WorkerCount = 0;
+
+	for (const FLayerInfo& LayerInfo : WorkerLayers)
+	{
+		check(*LayerInfo.LoadBalanceStrategy != nullptr);
+		WorkerCount += GetDefault<UAbstractLBStrategy>(*LayerInfo.LoadBalanceStrategy)->GetMinimumRequiredWorkers();
+	}
+
+	return WorkerCount;
+}
+
+#if WITH_EDITOR
+void UAbstractSpatialMultiWorkerSettings::ValidateFirstLayerIsDefaultLayer()
+{
+	// We currently rely on this for rendering debug information.
+	WorkerLayers[0].Name = SpatialConstants::DefaultLayer;
```

Comment:
If the first layer is not editable, why even have it in the array ? It could be added by default when pulling the layer info from the class.

Reply:
Also could crash if the user empties the array.